packages/addons-v5/lib/addons_wait.js

```diff
@@ -1,26 +1,60 @@
+/* eslint-disable no-await-in-loop */
+
 'use strict'
 
 const cli = require('heroku-cli-util')
 
-module.exports = async function (api, addon, interval) {
-  const app = addon.app.name
-  const addonName = addon.name
+module.exports = {
+  waitForAddonProvisioning: async function (api, addon, interval) {
+    const app = addon.app.name
+    const addonName = addon.name
+
+    await cli.action(`Creating ${cli.color.addon(addon.name)}`, (async function () {
+      while (addon.state === 'provisioning') {
+        // eslint-disable-next-line no-promise-executor-return
+        await new Promise(resolve => setTimeout(resolve, interval * 1000))
+
+        addon = await api.request({
+          method: 'GET',
+          path: `/apps/${app}/addons/${addonName}`,
+          headers: {'Accept-Expansion': 'addon_service,plan'},
+        })
+      }
+
+      if (addon.state === 'deprovisioned') {
+        throw new Error(`The add-on was unable to be created, with status ${addon.state}`)
+      }
+    })())
+
+    return addon
+  },
 
-  await cli.action(`Creating ${cli.color.addon(addon.name)}`, (async function () {
-    while (addon.state === 'provisioning') {
-      await new Promise(resolve => setTimeout(resolve, interval * 1000))
+  waitForAddonDeprovisioning: async function (api, addon, interval) {
+    const app = addon.app.name
+    const addonName = addon.name
 
-      addon = await api.request({
-        method: 'GET',
-        path: `/apps/${app}/addons/${addonName}`,
-        headers: {'Accept-Expansion': 'addon_service,plan'},
-      })
-    }
+    await cli.action(`Destroying ${cli.color.addon(addon.name)}`, (async function () {
+      while (addon.state === 'deprovisioning') {
+        // eslint-disable-next-line no-promise-executor-return
+        await new Promise(resolve => setTimeout(resolve, interval * 1000))
 
-    if (addon.state === 'deprovisioned') {
-      throw new Error(`The add-on was unable to be created, with status ${addon.state}`)
-    }
-  })())
+        await api.request({
+          method: 'GET',
+          path: `/apps/${app}/addons/${addonName}`,
+          headers: {'Accept-Expansion': 'addon_service,plan'},
+        }).then(addonInfo => {
+          addon = addonInfo
```

Comment:
Please create a new var instead of overwriting `addon`.

Reply:
Again, reused because of the `while` loop exit condition.

Reply:
Fixed following your suggestion